client/server/server.js

```diff
@@ -40,13 +41,22 @@ connection.onInitialize((_params) => {
 documents.onDidChangeContent((change) => {
     validateTextDocument(change.document);
 });
+let engineModel;
 // hold the maxNumberOfProblems setting
 let maxNumberOfProblems;
+let pathToPredefinedObjectsYml;
 // The settings have changed. Is send on server activation
 // as well.
 connection.onDidChangeConfiguration((change) => {
     let settings = change.settings;
     maxNumberOfProblems = settings.yseopml.maxNumberOfProblems || 100;
+    pathToPredefinedObjectsYml = settings.yseopml.pathToPredefinedObjectsYml;
+    if (engineModel == null) {
+        engineModel = new EngineModel_1.EngineModel(pathToPredefinedObjectsYml, completionItems);
+    }
```

Comment:
You've got quite a lot of

```
}
else {
```

in your JS code. That's surprising.

Reply:
The JS is generated from the `*.ts` files.